src/components/AssignedUsers/AssignedUsersContainer.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  FormattedMessage,
+  useIntl
+} from 'react-intl';
+
+import {
+  Headline,
+  Badge,
+  Loading,
+  Accordion,
+} from '@folio/stripes/components';
+import { useCallout } from '@folio/stripes/core';
+
+import {
+  useAssignedUsers,
+  useAssignedUsersMutation
+} from './hooks';
+import AssignedUsersList from './AssignedUsersList';
+import { findObjectDifferences } from './utils';
+
+const AssignedUsersContainer = ({ permissionsSet, expanded, onToggle, tenantId }) => {
+  const callout = useCallout();
+  const intl = useIntl();
+
+  const { grantedTo, id: permissionSetId, permissionName } = permissionsSet;
+  const [grantedToIds, setGrantedToIds] = useState(grantedTo);
+
+  const { users, isLoading, refetch } = useAssignedUsers({ grantedToIds, permissionSetId, tenantId });
+  const { assignUsers, unassignUsers } = useAssignedUsersMutation({ permissionSetId, tenantId, permissionName, setGrantedToIds });
+
+  const handleMutationSuccess = () => {
+    refetch();
+    callout.sendCallout({
+      message: <FormattedMessage id="ui-users.permissions.assignUsers.actions.message.success" />,
+      type: 'success',
+    });
+  };
+
+  const handleMutationError = async ({ response, message }) => {
+    const errorMessageId = 'ui-users.permissions.assignUsers.actions.message.error';
+
+    const error = await response?.text();
+    const statusCode = response?.status;
+    let errorMessage = error || message;
+
+    if (statusCode === 403 || statusCode === 405) {
+      errorMessage = <FormattedMessage id="ui-users.permissions.assignUsers.actions.message.permission.error" />;
+    }
+
+    return callout.sendCallout({
+      message: intl.formatMessage({ id: errorMessageId }, { error: errorMessage }),
+      type: 'error',
+      timeout: 0,
+    });
+  };
+
+  const handleAssignUsers = async (selectedUsers) => {
+    const { added, removed } = findObjectDifferences(users, selectedUsers);
```

Comment:
Do you really need to compare whole users' objects to distinguish updates?

Reply:
Yes, since the find-user plugin return only a selected users list and I should identify removed and newly added user ids. Do you have something in your mind?

Reply:
I thought you could just work with users' ids lists instead of whole objects. 
`differenceBy` and `intersectionBy` should help.

Reply:
updated